packages/reports/addon/components/filter-builders/time-dimension.ts

```diff
@@ -0,0 +1,221 @@
+      const interval = Interval.parseFromStrings(startStr, endStr);
+      const end = interval.asMomentsForTimePeriod(dateTimePeriod).end.utc(true);
+      let intervalTimePeriod = intervalPeriodForGrain(dateTimePeriod);
+      const nonAllGrain = dateTimePeriod === 'all' ? 'day' : dateTimePeriod;
+
+      let intervalValue;
+      if (end.isSame(moment.utc(getFirstDayOfGrain(moment.utc(), nonAllGrain)))) {
+        // end is 'current', get lookback amount
+        intervalValue = interval.diffForTimePeriod(intervalTimePeriod);
+      } else {
+        intervalValue = 1;
+      }
+
+      if (dateTimePeriod === 'quarter') {
+        // round to quarter
+        const quarters = Math.max(Math.floor(intervalValue / MONTHS_IN_QUARTER), 1);
+        intervalValue = quarters * MONTHS_IN_QUARTER;
+      }
+
+      const dateTimePeriodLabel = intervalTimePeriod[0].toUpperCase();
+      return [`P${intervalValue}${dateTimePeriodLabel}`, 'current'];
+    } else if (newOperator === OPERATORS.since) {
+      const interval = Interval.parseFromStrings(startStr, endStr);
+      const { start } = interval.asMomentsForTimePeriod(dateTimePeriod);
+      return [start.utc(true).toISOString()];
+    } else if (newOperator === OPERATORS.before) {
+      const interval = Interval.parseFromStrings(startStr, endStr);
+      const { end } = interval.asMomentsForTimePeriod(dateTimePeriod);
+      return [
+        end
+          .utc(true)
+          .subtract(1, getPeriodForGrain(dateTimePeriod))
+          .toISOString()
+      ];
+    } else if (newOperator === OPERATORS.dateRange) {
+      const interval = Interval.parseFromStrings(startStr, endStr);
+      const { start, end } = interval.asMomentsForTimePeriod(dateTimePeriod);
+      return [start.utc(true).toISOString(), end.utc(true).toISOString()];
+    }
+    warn(`No operator was found for the values '${prevValues.join(',')}'`, {
+      id: 'time-dimension-filter-builder-no-operator'
+    });
+
+    return [];
+  }
+
+  /**
+   * Finds the appropriate interval operator to modify an existing interval
+   * @param interval - the interval to choose an operator for
+   * @returns the best supported operator for this interval
+   */
+  @computed('args.filter.{values,operator}', 'supportedOperators')
+  get selectedOperator(): InteralFilterBuilderOperators {
+    const { values, operator } = this.args.filter;
+    const [startStr, endStr] = values;
+    if (!(startStr && endStr)) {
+      const filterValueBuilder = this.supportedOperators.find(({ id }) => id === operator);
+      assert(`A filter value component for ${operator} operator exists`, filterValueBuilder);
+      return filterValueBuilder;
+    }
+    const interval = Interval.parseFromStrings(startStr, endStr);
+    const { start, end } = interval.asStrings();
+
+    const [lookbackDuration, lookbackGrain] = parseDuration(start) || [];
+    let operatorId: InternalOperatorType;
+    if (start === 'current' && end === 'next') {
```

Comment:
How well does this work with Elide since between is inclusive/inclusive?

Reply:
I've updated this so that it looks right in the calendar and response, but kept the same behavior where date filter values are stored as `[inclusive, exclusive)`